op-service/txmgr/txmgr.go

```diff
@@ -914,16 +914,25 @@ func (m *SimpleTxManager) SuggestGasPriceCaps(ctx context.Context) (*big.Int, *b
 
 	// Enforce minimum base fee and tip cap
 	minTipCap := m.cfg.MinTipCap.Load()
+	maxTipCap := m.cfg.MaxTipCap.Load()
 	minBaseFee := m.cfg.MinBaseFee.Load()
+	maxBaseFee := m.cfg.MaxBaseFee.Load()
 
 	if minTipCap != nil && tip.Cmp(minTipCap) == -1 {
 		m.l.Debug("Enforcing min tip cap", "minTipCap", minTipCap, "origTipCap", tip)
 		tip = new(big.Int).Set(minTipCap)
 	}
+	if maxTipCap != nil && tip.Cmp(maxTipCap) > 0 {
+		return nil, nil, nil, fmt.Errorf("tip is too high: %v, cap:%v", tip, maxTipCap)
+	}
+
```

Comment:
The behaviour for max is quite different to min, i.e. when we are above the minimum, we clamp it to the minimum. But when we are above the max, we return an error. I think this would be worth reflecting in the documentation for the flags.

Reply:
And the godoc of this function, too.

Reply:
Done in 12da053f2b3e895fea804e805deadab3df28fb60.